src/Repository/ProgrammeRepository.php

```diff
@@ -77,20 +89,20 @@ public function getPaginated(int $page, int $limit): array
     public function getPaginatedFilteredSorted(
         array $paginate,
         array $filters,
-        string $sort,
-        string $direction
+        ?string $sort,
+        ?string $direction
     ): array {
         $query = $this->entityManager
             ->createQueryBuilder()
             ->select('p')
             ->from('App\Entity\Programme', 'p')
-            ->setFirstResult(($paginate['currentPage'] * $paginate['maxPerPage']) - $paginate['maxPerPage'])
-            ->setMaxResults($paginate['maxPerPage']);
+            ->setFirstResult($paginate['size'] * ($paginate['page'] - 1))
+            ->setMaxResults($paginate['size']);
 
         foreach ($filters as $key => $value) {
-            if ('' != $value) {
-                $query = $query->where("p.$key = :$key");
-                $query->setParameter(':key', $value);
+            if (in_array($key, self::PROGRAMME_FIELDS) && null != $value) {
+                $query->andWhere("p.$key LIKE :value");
+                $query->setParameter(':value', '%' . $value . '%');
```

Comment:
The % and % should be put in the line above IMO.

Reply:
I don't like the way it looks too, but I tried to place the '%' in line 104 but i recieve an error..